src/Microsoft.Azure.SignalR/HubHost/ServiceConnectionContext.cs

```diff
@@ -142,14 +143,18 @@ private HttpContext BuildHttpContext(OpenConnectionMessage message)
             var httpContextFeatures = new FeatureCollection();
             httpContextFeatures.Set<IHttpRequestFeature>(new HttpRequestFeature
             {
-                Headers = new HeaderDictionary((Dictionary<string, StringValues>) message.Headers),
+                Headers = new HeaderDictionary((Dictionary<string, StringValues>)message.Headers),
                 QueryString = message.QueryString,
                 Path = GetOriginalPath(message.QueryString)
             });
             httpContextFeatures.Set<IHttpAuthenticationFeature>(new HttpAuthenticationFeature
             {
                 User = User
             });
+            httpContextFeatures.Set<IHttpConnectionFeature>(new HttpConnectionFeature
+            {
+                RemoteIpAddress = IPAddress.Parse(message.Headers["X-Real-IP"][0])
```

Comment:
There's a standard header that reverse proxies normally add for the client IP https://en.wikipedia.org/wiki/X-Forwarded-For